model/action.go

```diff
@@ -0,0 +1,1070 @@
+	ActionTypeGoTo3DView  PdfActionType = "GoTo3DView"  // Set the current view of a 3D annotation
+	ActionTypeGoToE       PdfActionType = "GoToE"       // Go to embedded, PDF 1.6, Got to a destination in an embedded file
+	ActionTypeGoToR       PdfActionType = "GoToR"       // Go to remote, Go to a destination in another document
+	ActionTypeHide        PdfActionType = "Hide"        // Set an annotation's Hidden flag
+	ActionTypeImportData  PdfActionType = "ImportData"  // Import field values from a file
+	ActionTypeJavaScript  PdfActionType = "JavaScript"  // Execute a JavaScript script
+	ActionTypeLaunch      PdfActionType = "Launch"      // Launch an application, usually to open a file
+	ActionTypeMovie       PdfActionType = "Movie"       // Play a movie
+	ActionTypeNamed       PdfActionType = "Named"       // Execute an action predefined by the conforming reader
+	ActionTypeRendition   PdfActionType = "Rendition"   // Controls the playing of multimedia content
+	ActionTypeResetForm   PdfActionType = "ResetForm"   // Set fields to their default values
+	ActionTypeSetOCGState PdfActionType = "SetOCGState" // Set the states of optional content groups
+	ActionTypeSound       PdfActionType = "Sound"       // Play a sound
+	ActionTypeSubmitForm  PdfActionType = "SubmitForm"  // Send data to a uniform resource locator
+	ActionTypeThread      PdfActionType = "Thread"      // Begin reading an article thread
+	ActionTypeTrans       PdfActionType = "Trans"       // Updates the display of a document, using a transition dictionary
+	ActionTypeURI         PdfActionType = "URI"         // Resolves a uniform resource identifier
+)
+
+// PdfAction represents an action in PDF (section 12.6 p. 412).
+type PdfAction struct {
+	context PdfModel
+
+	Type core.PdfObject
+	S    core.PdfObject
+	Next core.PdfObject
+
+	container *core.PdfIndirectObject
+}
+
+// GetContext returns the action context which contains the specific type-dependent context.
+// The context represents the subaction.
+func (a *PdfAction) GetContext() PdfModel {
+	if a == nil {
+		return nil
+	}
+	return a.context
+}
+
+// SetContext sets the sub action (context).
+func (a *PdfAction) SetContext(ctx PdfModel) {
+	a.context = ctx
+}
+
+// GetContainingPdfObject implements interface PdfModel.
+func (a *PdfAction) GetContainingPdfObject() core.PdfObject {
+	return a.container
+}
+
+// ToPdfObject implements interface PdfModel.
+func (a *PdfAction) ToPdfObject() core.PdfObject {
+	container := a.container
+	d := container.PdfObject.(*core.PdfObjectDictionary)
+
+	d.Clear()
+
+	d.Set("Type", core.MakeName("Action"))
+	d.SetIfNotNil("S", a.S)
+	d.SetIfNotNil("Next", a.Next)
+
+	return container
+}
+
+// String implements interface PdfObject.
+func (a *PdfAction) String() string {
+	s := ""
+
+	obj, ok := a.ToPdfObject().(*core.PdfIndirectObject)
+	if ok {
+		s = fmt.Sprintf("%T: %s", a.context, obj.PdfObject.String())
+	}
+
+	return s
+}
+
+// PdfActionGoTo represents a GoTo action
```

Comment:
For consistency, should have `.` at the end of type comments.  Please address the rest of the action type structs as well.